### PR TITLE
Added a name to the fluent parameters

### PR DESCRIPTION
**Pull request by Framba-Luca**

### CODE REVIEW

File: unified_planning/model/problem.py [hunk 1]
```diff
@@ -428,7 +428,7 @@ def _get_ith_fluent_exp(self, fluent: 'up.model.fluent.Fluent', domain_sizes: Li
             ds = domain_sizes[i];
             rem = quot % ds
             quot //= ds
-            v = domain_item(self, fluent.signature()[i], rem)
+            v = domain_item(self, list(fluent.signature().values())[i], rem) #TODO: find a better way
```
Comment (mikand) [inline unified_planning/model/problem.py:431]: It's pretty easy: in line 427 you can simply iterate on the values ```for i, t in enumerate(fluent.signature().values()):```

File: unified_planning/test/examples/minimals.py [hunk 2]
```diff
@@ -169,7 +169,7 @@ def get_example_problems():
     # basic exists
     sem = UserType("Semaphore")
     x = Fluent('x')
-    y = Fluent('y', BoolType(), [sem])
+    y = Fluent('y', BoolType(), p1=sem)
```
Comment (mikand) [inline unified_planning/test/examples/minimals.py:172]:
```suggestion
    y = Fluent('y', BoolType(), semaphore=sem)
```

File: unified_planning/test/examples/minimals.py [hunk 3]
```diff
@@ -217,9 +217,9 @@ def get_example_problems():
 
     # temporal conditional
     Obj = UserType('Obj')
-    is_same_obj = Fluent('is_same_obj', BoolType(), [Obj, Obj])
-    is_ok = Fluent('is_ok', BoolType(), [Obj])
-    is_ok_giver = Fluent('is_ok_giver', BoolType(), [Obj])
+    is_same_obj = Fluent('is_same_obj', BoolType(), p1=Obj, p2=Obj)
```
Comment (mikand) [inline unified_planning/test/examples/minimals.py:220]:
```suggestion
    is_same_obj = Fluent('is_same_obj', BoolType(), object1=Obj, object2=Obj)
```

File: unified_planning/test/examples/minimals.py (hunk 2 above)
Comment (mikand) [reply unified_planning/test/examples/minimals.py:172]: Also in other cases, use meaningful names now that we have them

File: unified_planning/test/examples/realistic.py [hunk 4]
```diff
@@ -24,7 +24,7 @@ def get_example_problems():
 
     # robot
     Location = UserType('Location')
-    robot_at = Fluent('robot_at', BoolType(), [Location])
+    robot_at = Fluent('robot_at', BoolType(), p1=Location)
```
Comment (mikand) [inline unified_planning/test/examples/realistic.py:27]:
```suggestion
    robot_at = Fluent('robot_at', BoolType(), pos=Location)
```

Comment (mikand) [reply unified_planning/test/examples/realistic.py:27]: Same below

File: unified_planning/test/examples/testing_variants.py [hunk 5]
```diff
@@ -25,7 +25,7 @@ def get_example_problems():
     # robot_real_constants
     #this version of the problem robot has reals instead of integers as constants
     Location = UserType('Location')
-    robot_at = Fluent('robot_at', BoolType(), [Location])
+    robot_at = Fluent('robot_at', BoolType(), p1=Location)
```
Comment (mikand) [inline unified_planning/test/examples/testing_variants.py:28]: Same as before, use meaningful names

File: unified_planning/test/test_grounder.py [hunk 6]
```diff
@@ -162,7 +162,7 @@ def test_timed_connected_locations(self):
     def test_ad_hoc_1(self):
         problem = Problem('ad_hoc')
         Location = UserType('Location')
-        visited = Fluent('at', BoolType(), [Location])
+        visited = Fluent('at', BoolType(), p1=Location)
```
Comment (mikand) [inline unified_planning/test/test_grounder.py:165]: Same

File: unified_planning/model/fluent.py [hunk 7]
```diff
@@ -20,23 +20,34 @@
...
 from unified_planning.environment import get_env, Environment
-from typing import List
+from typing import List, OrderedDict, Optional
 
 
 class Fluent:
     """Represents a fluent."""
     def __init__(self, name: str, typename: 'unified_planning.model.types.Type' = None,
-                 signature: List['unified_planning.model.types.Type'] = [], env: Environment = None):
+                 signature: Optional[OrderedDict[str, 'unified_planning.model.types.Type']] = None, env: Environment = None, **kwargs):
```
Comment (alvalentini) [inline unified_planning/model/fluent.py:29]:
```suggestion
                 parameters: Optional[OrderedDict[str, 'unified_planning.model.types.Type']] = None, env: Environment = None, **kwargs: 'unified_planning.model.types.Type'):
```

**Review by mikand — CHANGES_REQUESTED** [review]
@Framba-Luca I left some comments to be fixed, but overall it looks good!

**Review by alvalentini — CHANGES_REQUESTED** [review]

**Review by alvalentini — APPROVED** [review]

**Pull request merged**